lib/elixir/src/elixir.erl

```diff
@@ -123,14 +123,18 @@ check_endianness() ->
   case code:ensure_loaded(?system) of
     {module, ?system} ->
       Endianness = ?system:endianness(),
-      case ?system:compiled_endianness() of
+      CompiledEndianness = ?system:compiled_endianness(),
+      case CompiledEndianness of
         Endianness ->
           ok;
         _ ->
           io:format(standard_error,
-            "warning: Elixir is running in a system with a different endianness than the one its "
-            "source code was compiled in. Please make sure Elixir and all source files were compiled "
-            "in a machine with the same endianness as the current one: ~ts~n", [Endianness])
+            "warning: Elixir is running in a system with a different endianness "
+            "than the one its source code was compiled in: ~ts "
+            "This may impact regex matching performance. "
+            "You may want to recompile elixir and and source files in a machine with "
+            "the same endianness as the current one: ~ts~n",
+            [CompiledEndianness, Endianness])
```

Comment:
I would prefer for the warning to be kept the same, given we provide a way for you to disable it. If you are running Elixir in production and you see this warning, then you want to act on it (and it should be scary). It may have other implications besides only regexes.

Reply:
Sure, I can revert this.
I had an impression that regexes were the only implication. Do you know any examples?

Reply:
I don't know any but I don't want to pass the impression there isn't any other, if that makes any sense. :)